src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/SNSAppointmentServiceTest.kt

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service
+
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import org.junit.jupiter.api.Test
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.component.SNSPublisher
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.EventDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.AppointmentEvent
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.AppointmentEventType
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Attended
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.SampleData
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.ActionPlanFactory
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.ReferralFactory
+import java.time.OffsetDateTime
+import java.util.UUID
+
+internal class SNSAppointmentServiceTest {
+  private val publisher = mock<SNSPublisher>()
+  private val snsAppointmentService = SNSAppointmentService(publisher)
+
+  private val actionPlan = ActionPlanFactory().create(
+    referral = ReferralFactory().createSent(id = UUID.fromString("56b40f96-0657-4e01-925c-da208a6fbcfd"))
+  )
+  private val now = OffsetDateTime.now()
+  private val attendanceRecordedEvent = AppointmentEvent(
+    "source",
+    AppointmentEventType.ATTENDANCE_RECORDED,
+    SampleData.sampleActionPlanAppointment(
+      actionPlan = actionPlan,
+      createdBy = actionPlan.createdBy,
+      attended = Attended.YES,
+      attendanceSubmittedAt = now,
+    ),
+    "http://localhost:8080/action-plan/77df9f6c-3fcb-4ec6-8fcf-96551cd9b080/session/1"
+  )
+
+  @Test
+  fun `appointment attendance recorded event publishes message with valid DTO`() {
+    snsAppointmentService.onApplicationEvent(attendanceRecordedEvent)
+
+    val eventDTO = EventDTO(
+      eventType = "intervention.session-appointment.attended",
```

Comment:
Should this also test the missed branch?

Reply:
yes probably. will add.